File: TsinghuaNet.UWP/ViewModels/DetailViewModel.vb
Public Class DetailViewModel
    Inherits TsinghuaNet.Helper.DetailViewModel

    Private details As List(Of NetDetail)

    Private _DetailsSource As IEnumerable(Of NetDetail)
    Public Property DetailsSource As IEnumerable(Of NetDetail)
        Get
            Return _DetailsSource
        End Get
        Set(value As IEnumerable(Of NetDetail))
            SetProperty(_DetailsSource, value)
        End Set
    End Property

    Protected Overrides Property InitialDetails As IEnumerable(Of NetDetail)
        Get
            Return details
        End Get
        Set(value As IEnumerable(Of NetDetail))
            details = value
            DetailsSource = details
        End Set
    End Property

    Protected Overrides Sub SetSortedDetails(source As IEnumerable(Of NetDetail))
        DetailsSource = source
    End Sub
End Class

File: TsinghuaNet.CrossPlatform/TsinghuaNet.CrossPlatform/ViewModels/DetailViewModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TsinghuaNet.CrossPlatform.ViewModels
{
    public class DetailViewModel : Helper.DetailViewModel
    {
        private IEnumerable<NetDetail> details;

        protected override IEnumerable<NetDetail> InitialDetails
        {
            get => details;
            set
            {
                details = value;
                DetailsSource = details;
            }
        }

        private IEnumerable<NetDetail> detailsSource;
        public IEnumerable<NetDetail> DetailsSource
        {
            get => detailsSource;
            set => SetProperty(ref detailsSource, value);
        }

        protected override void SetSortedDetails(IEnumerable<NetDetail> source) => DetailsSource = source;
    }
}
